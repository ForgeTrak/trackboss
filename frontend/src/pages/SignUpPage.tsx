import React from 'react';
import { useParams } from 'react-router-dom';
import { ChakraProvider } from '@chakra-ui/react';
import Header from '../components/Header';
import SignUpList from '../components/SignUpList';
import theme from '../theme';

type pageParams = {
    date: string,
};

export default function SignUpPage() {
    const { date } = useParams<pageParams>();
    return (
        <ChakraProvider theme={theme}>
            <Header title={`Sign Ups (${date})`} activeButtonId={2} />
            <SignUpList date={date} />
        </ChakraProvider>
    );
}
