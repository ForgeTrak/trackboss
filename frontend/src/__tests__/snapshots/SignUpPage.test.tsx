import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer, { act, ReactTestRenderer } from 'react-test-renderer';
import SignUpPage from '../../pages/SignUpPage';

describe('SignUp page', () => {
    it('renders correctly', () => {
        let list: ReactTestRenderer;
        act(() => {
            list = renderer.create(
                <BrowserRouter>
                    <SignUpPage />
                </BrowserRouter>,
            );
        });
        expect(list!.toJSON()).toMatchSnapshot();
    });
});
