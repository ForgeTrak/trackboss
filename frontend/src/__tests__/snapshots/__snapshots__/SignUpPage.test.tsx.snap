// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp page renders correctly 1`] = `
Array [
  <div>
    <div
      className="css-arm2n8"
    >
      <div
        className="chakra-stack css-84zodg"
      >
        <button
          aria-label="Back"
          className="chakra-button css-76fq9n"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable={false}
            height="lg"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="lg"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
            />
          </svg>
        </button>
        <div>
          <button
            aria-label="Menu"
            className="chakra-button css-1y85x9m"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden={true}
              fill="currentColor"
              focusable={false}
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        className="css-17xejub"
      />
      <div
        className="css-0"
      >
        <h2
          className="chakra-heading css-1ht9xq7"
        >
          Sign Ups (undefined)
        </h2>
      </div>
      <div
        className="css-17xejub"
      />
    </div>
  </div>,
  <div
    className="css-1bd5x9g"
  >
    <div
      data-testid="table"
    >
      <div
        className="css-gmuwbf"
      >
        <div
          className="css-1k55a1q"
        >
          <div
            className="css-0"
          >
            <div
              className="chakra-input__group css-4302v8"
            >
              <div
                className="chakra-input__left-element css-1cw84h2"
              >
                <svg
                  color="gray.300"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": "gray.300",
                    }
                  }
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
                  />
                </svg>
              </div>
              <input
                className="chakra-input css-1c4xw8j"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Search..."
              />
            </div>
          </div>
          <div
            className="css-thvrsi"
          >
            <button
              aria-label="Print"
              className="chakra-button css-ah3rcr"
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden={true}
                fill="currentColor"
                focusable={false}
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.5 8a.5.5 0 1 0 0-1 .5.5 0 0 0 0 1z"
                />
                <path
                  d="M5 1a2 2 0 0 0-2 2v2H2a2 2 0 0 0-2 2v3a2 2 0 0 0 2 2h1v1a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2v-1h1a2 2 0 0 0 2-2V7a2 2 0 0 0-2-2h-1V3a2 2 0 0 0-2-2H5zM4 3a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1v2H4V3zm1 5a2 2 0 0 0-2 2v1H2a1 1 0 0 1-1-1V7a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v3a1 1 0 0 1-1 1h-1v-1a2 2 0 0 0-2-2H5zm7 2v3a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1v-3a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        className="sc-dmRaPn gelpCx"
      >
        <div
          className="sc-fLlhyt ifOHjV"
        >
          <div
            className="sc-bczRLJ eSTlnH rdt_Table"
            disabled={false}
            role="table"
          >
            <div
              className="sc-ivTmOn fwKvpK"
            >
              <div
                style={
                  Object {
                    "padding": "24px",
                  }
                }
              >
                There are no records to display
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;
